scipy/optimize/__init__.py

```diff
@@ -86,6 +86,7 @@
 
    leastsq - Minimize the sum of squares of M equations in N unknowns
    nnls - Linear least-squares problem with non-negativity constraint
+   lsq_linear - Linear least-squares problem with arbitrary bound constraints
```

Comment:
I'd go with either "box constraints" or "bounded linear least squares". Now that I read the second option, I like it less.